src/ModelContextProtocol/ProgressNotificationValue.cs

```diff
@@ -1,7 +1,7 @@
 namespace ModelContextProtocol;
 
 /// <summary>Provides a progress value that can be sent using <see cref="IProgress{ProgressNotificationValue}"/>.</summary>
-public record struct ProgressNotificationValue
+public class ProgressNotificationValue
```

Comment:
In the same vein, can we also revisit `init` on these types? When they were records, `init` isn't terrible because you can at least use `with`. But as `class`, `init` is so much harder to work with. In most of these cases, there's not an obvious reason to me why they should be immutable.

Reply:
I personally prefer to keep them immutable unless there's an explicit need for them to be settable. I won't die on this hill though.